emily/lambda/src/main.rs

```diff
@@ -0,0 +1,175 @@
+                }),
+                last_update_block_hash: None, // Unknown on creation.
+                last_update_height: None, // Unknown on creation.
+                fulfillment: None, // Unknown on creation.
+            };
+
+            // Serialize response.
+            let serialized_response: String = serde_json::to_string(&response)
+                .map_err(Error::from)
+                .unwrap();
+
+            // Setup CORS headers.
+            // TODO: Refactor ApiGatewayProxyResponse packaging into a helper function.
+            let mut headers = HeaderMap::new();
+            headers.insert(header::CONTENT_TYPE, HeaderValue::from_static("application/json"));
+            headers.insert(header::ACCESS_CONTROL_ALLOW_HEADERS, HeaderValue::from_static("Content-Type"));
+            headers.insert(header::ACCESS_CONTROL_ALLOW_ORIGIN, HeaderValue::from_static("*"));
+            headers.insert(header::ACCESS_CONTROL_ALLOW_METHODS, HeaderValue::from_static("OPTIONS,POST,GET"));
+            Ok(ApiGatewayProxyResponse {
+                status_code: 201,
+                multi_value_headers: headers.clone(),
+                is_base64_encoded: false,
+                body: Some(Body::Text(serialized_response)),
+                headers,
+            })
+        }
+        _ => {
+            // Generate Not implemented for all unmatched requests.
+            let response: emily::models::NotImplementedErrorResponseContent = emily::models::NotImplementedErrorResponseContent {
+                message: "API call not implemented.".to_string()
+            };
+
+            // Serialize response.
+            let serialized_response: String = serde_json::to_string(&response)
+                .map_err(Error::from)
+                .unwrap();
+
+            // Setup CORS headers.
+            // TODO: Refactor ApiGatewayProxyResponse packaging into a helper function.
+            let mut headers = HeaderMap::new();
+            headers.insert(header::CONTENT_TYPE, HeaderValue::from_static("application/json"));
+            headers.insert(header::ACCESS_CONTROL_ALLOW_HEADERS, HeaderValue::from_static("Content-Type"));
+            headers.insert(header::ACCESS_CONTROL_ALLOW_ORIGIN, HeaderValue::from_static("*"));
+            headers.insert(header::ACCESS_CONTROL_ALLOW_METHODS, HeaderValue::from_static("OPTIONS,POST,GET"));
+            Ok(ApiGatewayProxyResponse {
+                status_code: 501,
+                multi_value_headers: headers.clone(),
+                is_base64_encoded: false,
+                body: Some(Body::Text(serialized_response)),
+                headers,
+            })
+        },
+    }
+}
+
+/// Main entry point for the AWS Lambda function.
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    // Run the lambda service.
+    lambda_runtime::run(
+        service_fn(
+            |event: LambdaEvent<ApiGatewayProxyRequest>| handle_event(event)
+        )
+    ).await
+}
```

Comment:
Nit (code organization): I'd like this to be the first defined method. In general I find code to be more readable if functions are ordered after their occurrence with the most significant functions at the top, and utility functions at the bottom.

Reply:
Done.

Reply:
:pray: